projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/DocumentService.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.service
+
+import org.springframework.core.io.Resource
+import org.springframework.http.ContentDisposition
+import org.springframework.http.HttpHeaders.CONTENT_DISPOSITION
+import org.springframework.http.HttpHeaders.CONTENT_LENGTH
+import org.springframework.http.HttpHeaders.CONTENT_TYPE
+import org.springframework.http.HttpHeaders.ETAG
+import org.springframework.http.HttpHeaders.LAST_MODIFIED
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.integrations.alfresco.AlfrescoClient
+import uk.gov.justice.digital.hmpps.integrations.delius.document.DocumentRepository
+import kotlin.text.Charsets.UTF_8
+
+@Service
+class DocumentService(
+    private val documentRepository: DocumentRepository,
+    private val alfrescoClient: AlfrescoClient
+) {
+    fun downloadDocument(crn: String, id: String): ResponseEntity<Resource> {
+        val filename = documentRepository.findNameByPersonCrnAndAlfrescoId(crn, id)
+            ?: throw NotFoundException("Document with id of $id not found for CRN $crn")
+
+        val alfrescoResponse = alfrescoClient.getDocument(id)
+        if (!alfrescoResponse.statusCode.is2xxSuccessful) {
+            throw RuntimeException("Failed to download document. Alfresco responded with ${alfrescoResponse.statusCode}.")
+        }
+
+        return ResponseEntity.ok()
```

Comment:
I thought the same thing after following the approach in the workforce integration: https://github.com/ministryofjustice/hmpps-probation-integration-services/blob/main/projects/workforce-allocations-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/DocumentService.kt.  It's a bit awkward because we're passing through HTTP headers from Alfresco.  I could've created a new class containing the headers and body to pass back to the controller, but it would've just ended up looking like a ResponseEntity anyway.